sentinel/src/main/kotlin/com/infinum/sentinel/ui/certificates/observer/SentinelWorkManager.kt

```diff
@@ -37,35 +39,38 @@ internal class SentinelWorkManager(
     }
 
     init {
-        WorkManager.initialize(
-            context,
-            Configuration.Builder()
-                .setMinimumLoggingLevel(android.util.Log.INFO)
-                .setWorkerFactory(workerFactory)
-                .build()
-        )
+        if (WorkManager.isInitialized().not()) {
+            WorkManager.initialize(
+                context,
+                Configuration.Builder()
+                    .setMinimumLoggingLevel(android.util.Log.INFO)
+                    .setWorkerFactory(workerFactory)
```

Comment:
What WorkerFactory do we inject here?

Reply:
`SentinelWorkerFactory` which knows how to create `CertificateCheckWorker`
But, now actually you are mentioning it, this shouldn't be needed anymore

Reply:
Let's remove it then. I'd remove the logging as well, or set it conditionally only for debug 🤷‍♂️

Reply:
If I remove debugging it seems we don't need to call `WorkManager.initialize` and then we can also `WorkManagerInitializer` from Manifest

Reply:
Does this also work when the host application doesn't initialize the work manager?